Tests/Integration/Network/Connection/FogSyncErrorConnectionIntTests.swift

```diff
@@ -0,0 +1,266 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import XCTest
+@testable import MobileCoin
+
+class FogSyncErrorConnectionIntTests: XCTestCase {
+    func testFogConsensusSame() throws {
+        try testSupportedProtocols(description: description) {
+            try fogConsensusSame(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogConsensusSame(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 61
+        let consensusIndex = fogIndex
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogAheadConsensus() throws {
+        try testSupportedProtocols(description: description) {
+            try fogAheadConsensus(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogAheadConsensus(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 321
+        let consensusIndex = fogIndex - 2
+        
+        // Fog ahead of Consensus should succeed (occurs when cached Consensus block info is used)
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogBehindConsensusWithinThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusWithinThresholdAheadConsensus(
+                    transportProtocol: $0,
+                    expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusWithinThresholdAheadConsensus(
```

Comment:
is there supposed to be the "AheadConsensus" suffix to this name?

Reply:
```suggestion
    func fogBehindConsensusButWithinThreshold(
```